datashader/utils.py

```diff
@@ -95,22 +104,22 @@ def calc_bbox(xs, ys, res):
         Two-tuple (int, int) which includes x and y resolutions (aka "grid/cell
         sizes"), respectively.
     """
+    xbound = xs.max() if res[0] < 0 else xs.min()
+    ybound = ys.min() if res[1] < 0 else ys.max()
+
     xmin = ymin = np.inf
     xmax = ymax = -np.inf
-    Ab = np.array([[res[0], 0.,      xs.min()],
-                   [0.,     -res[1], ys.max()],
-                   [0.,     0.,      1.]])
+    Ab = np.array([[res[0],  0.,      xbound],
+                   [0.,      -res[1], ybound],
+                   [0.,      0.,      1.]])
     for x_, y_ in [(0, 0), (0, len(ys)), (len(xs), 0), (len(xs), len(ys))]:
         x, y, _ = np.dot(Ab, np.array([x_, y_, 1.]))
-        if x < xmin:
-            xmin = x
-        if x > xmax:
-            xmax = x
-        if y < ymin:
-            ymin = y
-        if y > ymax:
-            ymax = y
-    return xmin, ymin, xmax, ymax
+        if x < xmin: xmin = x
+        if x > xmax: xmax = x
+        if y < ymin: ymin = y
+        if y > ymax: ymax = y
+    xpad, ypad = res[0]/2., res[1]/2.
```

Comment:
Is the half-grid adjustment still necessary, since we are now relying on xarray to provide bin-centered coordinates?